### DISCUSSION TITLE
Points' colors meaning in residuals vs linear predictors

### DISCUSSION THREAD

**Discussion by zohrazahir** [JenniNiku/gllvm#122] [Q&A] [answered]
Hello, 

### I am wondering if anyone knows what colors in the residuals vs linear predictor plot signify.

Here is the code for plotting: 

```
par(mfrow = c(1, 2))
plot(fit_gllvm, which = 1:2, n.plot = 100)
```

### When I add var.colors to the above code, it changes all the points to one color (for example, color 7, which is yellow) 

```

par(mfrow = c(1, 2))
plot(fit_gllvm, which = 1:2, n.plot = 100, var.colors = 7)
```
![without var colors](https://github.com/JenniNiku/gllvm/assets/72106136/f9405fa7-5815-4103-a79a-5c306349c158)
![with var colors](https://github.com/JenniNiku/gllvm/assets/72106136/298df8b0-e44b-42f1-b406-88e4c1710166)

Also, can you help me interpret the below graph? (It shows the residuals vs linear predictor based on Poisson and negative binomial: 

 
![Poisson vs NB](https://github.com/JenniNiku/gllvm/assets/72106136/9f5336d3-773b-4b39-8648-7a9344109e07)

Thank you,
Z.

**Comment by BertvanderVeen** [comment]
Thanks. I will respond to your question, but in the discussion forum as this is not a bug :)

**Reply by BertvanderVeen** [reply, marked as answer]
Hello! Each color represents the residuals that are associated with a separate species in your data. If you set the `var.colors` argument to a single value, then the residuals for all species will take that color.

With respect to the interpretation of your residual plot; residual plots can be tricky to interpret! Generally, the residual versus fitted plot should not exhibit any patterns. If you fit a Poisson model and there is a "fan-like" shape that is indicative of overdispersion, and you should switch to a model with a negative-binomial distribution. Here, your residuals plot for the Poisson model looks good, whereas that for the negative-binomial model looks a bit off; there are some values for a species that are predicted to be quite negative. However, the associated residual value for those fitted values do not look very extreme. It might be fine, but it does look a bit odd to me.
